--- origsrc/gnome-session-3.0.1/configure.ac	2011-04-04 16:15:41.000000000 -0500
+++ src/gnome-session-3.0.1/configure.ac	2011-05-13 16:46:12.842482800 -0500
@@ -56,10 +56,19 @@ PKG_CHECK_MODULES(GNOME_SESSION,
         gio-2.0 >= $GLIB_REQUIRED
         gtk+-3.0 >= $GTK3_REQUIRED
         dbus-glib-1 >= $DBUS_GLIB_REQUIRED
-        upower-glib >= $UPOWER_REQUIRED
         json-glib-1.0 >= $JSON_GLIB_REQUIRED
 )
 
+PKG_CHECK_MODULES(UPOWER,
+        upower-glib >= $UPOWER_REQUIRED,
+        have_upower=true,
+        have_upower=false
+)
+
+if test "$have_upower" = true; then
+  AC_DEFINE(WITH_UPOWER, 1, [Set to 1 if upower-glib is available.])
+fi
+
 PKG_CHECK_MODULES(SESSION_PROPERTIES,
         glib-2.0 >= $GLIB_REQUIRED
         gtk+-3.0 >= $GTK3_REQUIRED
--- origsrc/gnome-session-3.0.1/gnome-session/gsm-logout-dialog.c	2011-03-22 15:31:43.000000000 -0500
+++ src/gnome-session-3.0.1/gnome-session/gsm-logout-dialog.c	2011-05-13 16:48:12.608396400 -0500
@@ -27,7 +27,9 @@
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
 
+#ifdef WITH_UPOWER
 #include <upower.h>
+#endif
 
 #include "gsm-logout-dialog.h"
 #include "gsm-consolekit.h"
@@ -52,7 +54,9 @@ struct _GsmLogoutDialogPrivate
 {
         GsmDialogLogoutType  type;
 
+#ifdef WITH_UPOWER
         UpClient            *up_client;
+#endif
         GsmSystem           *system;
 
         int                  timeout;
@@ -144,7 +148,9 @@ gsm_logout_dialog_init (GsmLogoutDialog
         gtk_window_set_keep_above (GTK_WINDOW (logout_dialog), TRUE);
         gtk_window_stick (GTK_WINDOW (logout_dialog));
 
+#ifdef WITH_UPOWER
         logout_dialog->priv->up_client = up_client_new ();
+#endif
 
         logout_dialog->priv->consolekit = gsm_get_consolekit ();
 
@@ -168,10 +174,12 @@ gsm_logout_dialog_destroy (GsmLogoutDial
                 logout_dialog->priv->timeout_id = 0;
         }
 
+#ifdef WITH_UPOWER
         if (logout_dialog->priv->up_client) {
                 g_object_unref (logout_dialog->priv->up_client);
                 logout_dialog->priv->up_client = NULL;
         }
+#endif
 
         if (logout_dialog->priv->consolekit) {
                 g_object_unref (logout_dialog->priv->consolekit);
@@ -184,13 +192,21 @@ gsm_logout_dialog_destroy (GsmLogoutDial
 static gboolean
 gsm_logout_supports_system_suspend (GsmLogoutDialog *logout_dialog)
 {
+#ifdef WITH_UPOWER
         return up_client_get_can_suspend (logout_dialog->priv->up_client);
+#else
+        return FALSE;
+#endif
 }
 
 static gboolean
 gsm_logout_supports_system_hibernate (GsmLogoutDialog *logout_dialog)
 {
+#ifdef WITH_UPOWER
         return up_client_get_can_hibernate (logout_dialog->priv->up_client);
+#else
+        return FALSE;
+#endif
 }
 
 static gboolean
--- origsrc/gnome-session-3.0.1/gnome-session/gsm-manager.c	2011-03-30 02:47:33.000000000 -0500
+++ src/gnome-session-3.0.1/gnome-session/gsm-manager.c	2011-05-13 16:53:38.163147400 -0500
@@ -39,7 +39,9 @@
 #include <dbus/dbus-glib-lowlevel.h>
 #include <librsvg/rsvg.h>
 
+#ifdef WITH_UPOWER
 #include <upower.h>
+#endif
 
 #include <gtk/gtk.h> /* for logout dialog */
 
@@ -153,8 +155,10 @@ struct GsmManagerPrivate
         DBusGProxy             *bus_proxy;
         DBusGConnection        *connection;
 
+#ifdef WITH_UPOWER
         /* Interface with other parts of the system */
         UpClient               *up_client;
+#endif
 
         GsmShell               *shell;
         guint                   shell_end_session_dialog_canceled_id;
@@ -1096,6 +1100,7 @@ manager_perhaps_lock (GsmManager *manage
 static void
 manager_attempt_hibernate (GsmManager *manager)
 {
+#ifdef WITH_UPOWER
         gboolean  can_hibernate;
         GError   *error;
         gboolean  ret;
@@ -1114,11 +1119,13 @@ manager_attempt_hibernate (GsmManager *m
                         g_error_free (error);
                 }
         }
+#endif
 }
 
 static void
 manager_attempt_suspend (GsmManager *manager)
 {
+#ifdef WITH_UPOWER
         gboolean  can_suspend;
         GError   *error;
         gboolean  ret;
@@ -1137,6 +1144,7 @@ manager_attempt_suspend (GsmManager *man
                         g_error_free (error);
                 }
         }
+#endif
 }
 
 static void
@@ -2519,10 +2527,12 @@ gsm_manager_dispose (GObject *object)
                 manager->priv->lockdown_settings = NULL;
         }
 
+#ifdef WITH_UPOWER
         if (manager->priv->up_client != NULL) {
                 g_object_unref (manager->priv->up_client);
                 manager->priv->up_client = NULL;
         }
+#endif
 
         if (manager->priv->shell != NULL) {
                 g_object_unref (manager->priv->shell);
@@ -2728,7 +2738,9 @@ gsm_manager_init (GsmManager *manager)
                                       NULL,
                                       NULL, NULL);
 
+#ifdef WITH_UPOWER
         manager->priv->up_client = up_client_new ();
+#endif
 
         manager->priv->shell = gsm_get_shell ();
 }
@@ -3329,13 +3341,15 @@ gsm_manager_can_shutdown (GsmManager *ma
                           GError    **error)
 {
         GsmSystem *system;
-        gboolean can_suspend;
-        gboolean can_hibernate;
+        gboolean can_suspend = FALSE;
+        gboolean can_hibernate = FALSE;
 
+#ifdef WITH_UPOWER
         g_object_get (manager->priv->up_client,
                       "can-suspend", &can_suspend,
                       "can-hibernate", &can_hibernate,
                       NULL);
+#endif
 
         g_debug ("GsmManager: CanShutdown called");
 
