--- origsrc/gnome-session-3.0.1/configure.ac	2011-04-04 16:15:41.000000000 -0500
+++ src/gnome-session-3.0.1/configure.ac	2011-05-13 16:46:12.842482800 -0500
@@ -56,10 +56,19 @@ PKG_CHECK_MODULES(GNOME_SESSION,
         gio-2.0 >= $GLIB_REQUIRED
         gtk+-3.0 >= $GTK3_REQUIRED
         dbus-glib-1 >= $DBUS_GLIB_REQUIRED
-        upower-glib >= $UPOWER_REQUIRED
         json-glib-1.0 >= $JSON_GLIB_REQUIRED
 )
 
+PKG_CHECK_MODULES(UPOWER,
+        upower-glib >= $UPOWER_REQUIRED,
+        have_upower=true,
+        have_upower=false
+)
+
+if test "$have_upower" = true; then
+  AC_DEFINE(WITH_UPOWER, 1, [Set to 1 if upower-glib is available.])
+fi
+
 PKG_CHECK_MODULES(SESSION_PROPERTIES,
         glib-2.0 >= $GLIB_REQUIRED
         gtk+-3.0 >= $GTK3_REQUIRED
--- origsrc/gnome-session-3.0.1/gnome-session/gsm-logout-dialog.c	2011-03-22 15:31:43.000000000 -0500
+++ src/gnome-session-3.0.1/gnome-session/gsm-logout-dialog.c	2011-05-13 16:48:12.608396400 -0500
@@ -27,7 +27,9 @@
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
 
+#ifdef WITH_UPOWER
 #include <upower.h>
+#endif
 
 #include "gsm-logout-dialog.h"
 #include "gsm-consolekit.h"
@@ -52,7 +54,9 @@ struct _GsmLogoutDialogPrivate
 {
         GsmDialogLogoutType  type;
 
+#ifdef WITH_UPOWER
         UpClient            *up_client;
+#endif
         GsmSystem           *system;
 
         int                  timeout;
@@ -144,7 +148,9 @@ gsm_logout_dialog_init (GsmLogoutDialog
         gtk_window_set_keep_above (GTK_WINDOW (logout_dialog), TRUE);
         gtk_window_stick (GTK_WINDOW (logout_dialog));
 
+#ifdef WITH_UPOWER
         logout_dialog->priv->up_client = up_client_new ();
+#endif
 
         logout_dialog->priv->consolekit = gsm_get_consolekit ();
 
@@ -168,10 +174,12 @@ gsm_logout_dialog_destroy (GsmLogoutDial
                 logout_dialog->priv->timeout_id = 0;
         }
 
+#ifdef WITH_UPOWER
         if (logout_dialog->priv->up_client) {
                 g_object_unref (logout_dialog->priv->up_client);
                 logout_dialog->priv->up_client = NULL;
         }
+#endif
 
         if (logout_dialog->priv->consolekit) {
                 g_object_unref (logout_dialog->priv->consolekit);
@@ -184,13 +192,21 @@ gsm_logout_dialog_destroy (GsmLogoutDial
 static gboolean
 gsm_logout_supports_system_suspend (GsmLogoutDialog *logout_dialog)
 {
+#ifdef WITH_UPOWER
         return up_client_get_can_suspend (logout_dialog->priv->up_client);
+#else
+        return FALSE;
+#endif
 }
 
 static gboolean
 gsm_logout_supports_system_hibernate (GsmLogoutDialog *logout_dialog)
 {
+#ifdef WITH_UPOWER
         return up_client_get_can_hibernate (logout_dialog->priv->up_client);
+#else
+        return FALSE;
+#endif
 }
 
 static gboolean
--- origsrc/gnome-session-3.6.2/gnome-session/gsm-consolekit.c	2012-11-12 17:32:21.000000000 -0600
+++ src/gnome-session-3.6.2/gnome-session/gsm-consolekit.c	2012-12-19 19:48:14.730872800 -0600
@@ -31,7 +31,9 @@
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
 
+#ifdef WITH_UPOWER
 #include <upower.h>
+#endif
 
 #include "gsm-system.h"
 #include "gsm-consolekit.h"
@@ -55,7 +57,9 @@ struct _GsmConsolekitPrivate
         DBusGConnection *dbus_connection;
         DBusGProxy      *bus_proxy;
         DBusGProxy      *ck_proxy;
+#ifdef WITH_UPOWER
         UpClient        *up_client;
+#endif
 };
 
 static void     gsm_consolekit_class_init   (GsmConsolekitClass *klass);
@@ -214,6 +218,7 @@ gsm_consolekit_on_name_owner_changed (DB
                                       const char    *new_owner,
                                       GsmConsolekit *manager)
 {
+#ifdef WITH_UPOWER
         if (name != NULL && strcmp (name, "org.freedesktop.ConsoleKit") != 0) {
                 return;
         }
@@ -223,7 +228,7 @@ gsm_consolekit_on_name_owner_changed (DB
 
         gsm_consolekit_ensure_ck_connection (manager, NULL);
         manager->priv->up_client = up_client_new ();
-
+#endif
 }
 
 static void
@@ -247,7 +252,9 @@ gsm_consolekit_free_dbus (GsmConsolekit
 {
         g_clear_object (&manager->priv->bus_proxy);
         g_clear_object (&manager->priv->ck_proxy);
+#ifdef WITH_UPOWER
         g_clear_object (&manager->priv->up_client);
+#endif
 
         if (manager->priv->dbus_connection != NULL) {
                 DBusConnection *connection;
@@ -814,22 +821,31 @@ gsm_consolekit_is_login_session (GsmSyst
 static gboolean
 gsm_consolekit_can_suspend (GsmSystem *system)
 {
+#ifdef WITH_UPOWER
         GsmConsolekit *consolekit = GSM_CONSOLEKIT (system);
 
         return up_client_get_can_suspend (consolekit->priv->up_client);
+#else
+        return FALSE;
+#endif
 }
 
 static gboolean
 gsm_consolekit_can_hibernate (GsmSystem *system)
 {
+#ifdef WITH_UPOWER
         GsmConsolekit *consolekit = GSM_CONSOLEKIT (system);
 
         return up_client_get_can_hibernate (consolekit->priv->up_client);
+#else
+        return FALSE;
+#endif
 }
 
 static void
 gsm_consolekit_suspend (GsmSystem *system)
 {
+#ifdef WITH_UPOWER
         GsmConsolekit *consolekit = GSM_CONSOLEKIT (system);
         GError *error = NULL;
         gboolean ret;
@@ -839,11 +855,13 @@ gsm_consolekit_suspend (GsmSystem *syste
                 g_warning ("Unexpected suspend failure: %s", error->message);
                 g_error_free (error);
         }
+#endif
 }
 
 static void
 gsm_consolekit_hibernate (GsmSystem *system)
 {
+#ifdef WITH_UPOWER
         GsmConsolekit *consolekit = GSM_CONSOLEKIT (system);
         GError *error = NULL;
         gboolean ret;
@@ -853,6 +871,7 @@ gsm_consolekit_hibernate (GsmSystem *sys
                 g_warning ("Unexpected hibernate failure: %s", error->message);
                 g_error_free (error);
         }
+#endif
 }
 
 static void
